src/main/java/org/jboss/threads/StripedEnhancedQueueExecutor.java

```diff
@@ -0,0 +1,736 @@
+        int maxQueueSize = builder.maxQueueSize;
+        if (maxQueueSize != Integer.MAX_VALUE) {
+            maxQueueSize = (maxQueueSize % count == 0 ? 0 : 1) + maxQueueSize / count;
+        }
+        final List<EnhancedQueueExecutor> executors = new ArrayList<>(count);
+        for (int i = 0; i < count; i ++) {
+            EnhancedQueueExecutor.Builder b = new EnhancedQueueExecutor.Builder();
+            b.setTerminationTask(stripeStopTask);
+            b.setThreadFactory(builder.threadFactory);
+            b.setCorePoolSize(coreSize);
+            b.setMaximumPoolSize(maxSize);
+            b.setExceptionHandler(builder.exceptionHandler);
+            b.setGrowthResistance(builder.growthResistance);
+            b.allowCoreThreadTimeOut(builder.allowCoreTimeOut);
+            b.setMaximumQueueSize(maxQueueSize);
+            b.setHandoffExecutor(builder.handoffExecutor);
+            b.setMBeanName(builder.mBeanName + "_" + i);
+            b.setKeepAliveTime(builder.keepAliveTime);
+            executors.add(b.build());
+        }
+        this.executors = executors;
+    }
+
+    public StandardThreadPoolMXBean getThreadPoolMXBean() {
+        return new MXBean();
+    }
+
+    public int getStripes() {
+        return executors.size();
+    }
+
+    public void shutdown() {
+        for (EnhancedQueueExecutor executor : executors) {
+            executor.shutdown();
+        }
+    }
+
+    public List<Runnable> shutdownNow() {
+        List<Runnable> list = null;
+        for (EnhancedQueueExecutor executor : executors) {
+            List<Runnable> tasks = executor.shutdownNow();
+            if (! tasks.isEmpty()) {
+                if (list == null) {
+                    list = new ArrayList<>();
+                }
+                list.addAll(tasks);
+            }
+        }
+        return list == null ? Collections.emptyList() : list;
+    }
+
+    public boolean isShutdown() {
+        for (EnhancedQueueExecutor executor : executors) {
+            if (executor.isShutdown()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isTerminating() {
+        for (EnhancedQueueExecutor executor : executors) {
+            if (executor.isTerminating()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isTerminated() {
+        for (EnhancedQueueExecutor executor : executors) {
+            if (! executor.isTerminated()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean awaitTermination(final long timeout, final TimeUnit unit) throws InterruptedException {
+        long remaining = unit.toNanos(timeout);
+        long start = System.nanoTime();
+        long now;
+        for (EnhancedQueueExecutor executor : executors) {
+            executor.awaitTermination(remaining, TimeUnit.NANOSECONDS);
```

Comment:
If any `executor.awaitTermination` returns false, the final return value should be `false`, otherwise `true` (broke my benchmark harness which expects executors to terminate cleanly)

Reply:
Yeah you're right, I meant to do this but then got distracted by the "shiny object" of remaining timeout calculation :)